### ISSUE TITLE
Self-relation whereHas on descendants doesn't work

### ISSUE THREAD

**Issue by InToSSH** [lazychaser/laravel-nestedset#206]
Hello,
not sure if this is problem with the package or the Laravel itself.
I am using Laravel 5.2, updated to newest commit, including fix for: [This bug](https://github.com/laravel/framework/pull/12146/commits/cd9d45f0c44d470f96e0e12d617267a4ea042512)
I applied nestedset to Category model, and I need to create a scope to get only Categories which have some active Items or the descendants of the Category have some active Items.

For this I created a scope like this:
``` 
public function scopeHasActiveItems($query) {

        return $query->hasDirectActiveItems()->orWhereHas('descendants', function($query) {
            $query->hasDirectActiveItems();
        });
    }
```

Problem is that when I call this scope I get this SQL command:
```
select * from `categories` where `parent_id` is null and exists (select * from `categories` as `self_d85d7494de16c6ee89d5cd79eff9e843` where `self_d85d7494de16c6ee89d5cd79eff9e843`.`_lft` between `categories`.`_lft` + 1 and `categories`.`_rgt` and exists (select * from `items` where `items`.`category_id` = `categories`.`id` and `active` = ? and exists (select * from `item_variants` where `item_variants`.`item_id` = `items`.`id` and `active` = ?)))
```

But instead it should be:
```
select * from `categories` where `parent_id` is null and exists (select * from `categories` as `self_d85d7494de16c6ee89d5cd79eff9e843` where `self_d85d7494de16c6ee89d5cd79eff9e843`.`_lft` between `categories`.`_lft` + 1 and `categories`.`_rgt` and exists (select * from `items` where `items`.`category_id` = `self_d85d7494de16c6ee89d5cd79eff9e843`.`id` and `active` = ? and exists (select * from `item_variants` where `item_variants`.`item_id` = `items`.`id` and `active` = ?)))
```

Is there a way to fix this please?

**Comment by InToSSH** [comment]
Alright, I did some digging and it looks like you are missing:
`$query->getModel()->setTable($hash);`
in the DescendantsRelation.php file, method: getRelationExistanceQuery:

```
public function getRelationExistenceQuery(EloquentBuilder $query, EloquentBuilder $parent,
                                        $columns = [ '*' ]
    ) {
        $query->select($columns);

        $table = $query->getModel()->getTable();

        $query->from($table.' as '.$hash = $this->getRelationCountHash());
        $query->getModel()->setTable($hash);

        $grammar = $query->getQuery()->getGrammar();

        $table = $grammar->wrapTable($table);
        $hash = $grammar->wrapTable($hash);
        $lft = $grammar->wrap($this->parent->getLftName());
        $rgt = $grammar->wrap($this->parent->getRgtName());

        return $query->whereRaw("{$hash}.{$lft} between {$table}.{$lft} + 1 and {$table}.{$rgt}");
    }
```

**Comment by InToSSH** [comment]
#207 Added a pull request.

**Comment by lazychaser** [comment]
Fixed in v4.2.5
